maven/plugins/common/src/main/java/com/vmware/pscoe/maven/plugins/AbstractInstallNodeDepsMojo.java

```diff
@@ -80,10 +83,41 @@ public void execute() throws MojoExecutionException, MojoFailureException {
 				.execute(getLog());
         }
 
-        new ProcessExecutor()
-			.name("Dependency installation")
-			.directory(project.getBasedir())
-			.command(deps)
-			.execute(getLog());
+        getLog().info("deps length  " + deps.stream().mapToInt(String:: length).sum());
+
+		for(String path : deps)
+		{
+			commandLenght = commandLenght + path.length();
+			if(commandLenght <= 7000)
+			{
+				deps_cmd_7000.add(path);
+			}
+			else
+			{
+				
+				new ProcessExecutor()
+					.name("Dependency installation - Command Length is less than or equal 7000")
+					.directory(project.getBasedir())
+					.command(deps_cmd_7000)
+					.execute(getLog());
+
+				deps_cmd_7000 = new LinkedList<>();
+				deps_cmd_7000.add(npmExec);
+        		deps_cmd_7000.add("install");
+				deps_cmd_7000.add(path);
+				commandLenght = path.length();
+
+			}
+
+			if(deps.indexOf(path) == (deps.size() - 1))
```

Comment:
I would suggest to move this out of the loop, and then just check if (deps_cmd_7000.length() > 0) and then execute the command.

Reply:
Thank you for your review, will add this change